docs/spec_draft.md

```diff
@@ -1243,6 +1244,78 @@ Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.slice
+
+### Semantics
+
+Extracts a sub-tensor from the `operand` using the dimension indices from the
+operands.
+
+More formally, `result[i0, ..., ik, ..., iR-1] = operand[j0+s0, ..., jk+sk, ..., jR-1+sR-1]`
+for all `sk` in `strides` for all `jk` in `start_indices[k]` $\leq$ `jk` $\lt$
+`limit_indices[k]` where 0 $\leq$ `k` $\lt$ rank(`operand`).
+
+Shape is calculated by `dim(result, i)` = $\lceil$`(limit_indices[i]-start_indices[i])/stride[i]`$\rceil$
+for `i`th dimension size of `operand`.
+
+### Operands
+
+| Name            | Type                                                                                                                                    |
+|-----------------|-----------------------------------------------------------------------------------------------------------------------------------------|
+| `operand`       | tensor of any supported types                                                                                                           |
+| `start_indices` | 1-dimensional array of integers containing the starting indices of the slice for each dimension                                         |
+| `limit_indices` | 1-dimensional array of integers containing the ending indices (exclusive) for the slice for each dimension                              |
+| `strides`       | 1-dimensional array of integers that decides the input stride of the slice. The slice picks every `strides[d]` element in `d` dimension |
+
+### Results
+
+| Name     | Type                          |
+|----------|-------------------------------|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
+  * (C2) `start_indices`, `limit_indices`, and `strides` have the same type.
+  * (C3) size(`start_indices`) = size(`limit_indices`) = size(`strides`) =
+  rank(`operand`).
+  * (C4) 0 $\le$ `start_indices[d]` $\le$ `limit_indices[d]` $\le$
+  size(`operand[d]`) for all dimension `d`.
+  * (C5) 0 $\lt$ `strides[d]` for all dimension `d`.
+
+### Examples
+
+```mlir
+// 1-dimensional slice
+
+// %operand: [0, 1, 2, 3, 4]
+%result = "stablehlo.slice" (%operand) {
```

Comment:
Extraneous whitespace between the mnemonic and the open parenthesis. (Same below).